packages/core/src/builtin-tasks/moleculec.ts

```diff
@@ -0,0 +1,278 @@
+import { execSync } from 'node:child_process'
+import fs from 'node:fs'
+import path from 'node:path'
+import { task } from '../config/config-env'
+import { paramTypes } from '../params'
+
+interface Args {
+  schemaFile: string
+  language: 'rust' | 'c' | 'ts'
+  output?: string
+}
+
+const extMap: Record<Args['language'], string> = {
+  rust: '.rs',
+  c: '.cpp',
+  ts: '.ts',
+}
+
+const uintLength = [1, 2, 4, 6, 16, 32, 64]
+
+const basicMoleculeType = 'byte' as const
+
+const uintTypes = ['Uint8', 'Uint16', 'Uint32', 'Uint64', 'Uint128', 'Uint256', 'Uint512'] as const
+
+const tsMolecuelType = uintTypes[0]
+
+type SimpleMoleculeType = (typeof uintTypes)[number]
```

Comment:
`unitLength` is not continuously increasing integers, so I can not define `unitLength` and `unitTypes` in an Enum

Reply:
The Enum can be discrete as 
```typescript
enum UInts {
  UInt8 = 1,
  UInt16 = 2,
  UInt32 = 4,
  UInt64 = 8,
  Uint128 = 16,
  Uint256 = 32,
  Uint512 = 64,
}
```
Then 

```typescript
const uintLength = [1, 2, 4, 6, 16, 32, 64]

const uintTypes = ['Uint8', 'Uint16', 'Uint32', 'Uint64', 'Uint128', 'Uint256', 'Uint512'] as const

const tsMolecuelType = uintTypes[0]

type SimpleMoleculeType = (typeof uintTypes)[number]

const uintIdx = uintLength.indexOf(itemCount)
if (uintIdx !== -1 && uintTypes[uintIdx]) {
  return uintTypes[uintIdx]
}
```

can be transformed to 


```
const uintLength = UInts

const uintTypes = Object.values(UInts).filter(v => typeof v === 'string')

const tsMolecuelType = UInts[0]

type SimpleMoleculeType = UInts

const uintType = UInts[itemCount]

if (uintType) {
  return uintType
}
```

It can be further optimized based on their usage

BTW, `uintLength[3]` is set `6`, it should be `8`